### DISCUSSION TITLE
how to edit cmb files

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#622] [Help]
Originally posted 2019-12-31 17:40:45.183000 by jrc in GrandOrgue discussion forum on SourceForge<br><br>Hi.
I have been playing round modifying organs, with the object of putting my own sample set in instead of, for instance, the burea set. Editing the .organ files is no problem, just takes LOTS of time, but I have a problem with .cmb files.  Those I created some while ago are editable txt files, but recent ones seem to have changed format,  and are not editable with a text editor, and also seem to be a different structure....... HELP!!!!
john
ps - I am using Ubuntu linux

**Comment by larspalo** [comment]
Originally posted 2020-03-05 17:28:59.909000 by kerkovits in GrandOrgue discussion forum on SourceForge<br><br>I observed this issue, too. After investigating the binary, I found that the solution is just to decompress the file using gunzip.

P.S. I really miss a checkbox at the settings panel, if I want to have the cmb-files gzipped. It takes a lot of resources when GO decompresses the cmb, it eats my CPU significantly increasing the load time.

**Reply by larspalo** [reply]
Originally posted 2020-06-21 22:17:51.896000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>It would be really interessting to know, what is the size of your settings file and how much CPU usage increase are you noticing?

**Comment by larspalo** [comment]
Originally posted 2020-06-08 14:03:25.935000 by edwardstauff in GrandOrgue discussion forum on SourceForge<br><br>Could someone on the development team please answer this question (how to edit .cmd files), and add it to the documentation?
I need to examine .cmb files, too.  
I tried gunzip, but it just said &quot;unknown suffix -- ignored&quot;.

**Comment by larspalo** [comment]
Originally posted 2020-06-09 06:47:36.424000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>
I have tried several different unzip programs on my .cmb files and none of them will accept the files as valid .zip files. If these files are supposed to be normal .zip files then there is a bug in the GO zipper which produces corrupted files.

csw900

**Reply by larspalo** [reply]
Originally posted 2020-06-09 06:54:12.840000 by ggoode_sa in GrandOrgue discussion forum on SourceForge<br><br>I can read a .cmb file with a normal text reader (Notepad++) in
Windows. It&#039;s not compressed on my system.

On 6/9/20, csw900 &lt;csw900@users.sourceforge.net&gt; wrote:
&gt;
&gt; I have tried several different unzip programs on my .cmb files and none of
&gt; them will accept the files as valid .zip files. If these files are supposed
&gt; to be normal .zip files then there is a bug in the GO zipper which produces
&gt; corrupted files.
&gt;
&gt; csw900
&gt;
&gt;
&gt;
&gt; ---
&gt;
&gt; [how to edit cmb
&gt; files](https://sourceforge.net/p/ourorgan/discussion/962125/thread/e79cea8f67/?limit=25#7f7d)
&gt;
&gt;
&gt; ---
&gt;
&gt; Sent from sourceforge.net because you indicated interest in
&gt; &lt;https://sourceforge.net/p/ourorgan/discussion/962125/&gt;
&gt;
&gt;
&gt;
&gt; To unsubscribe from further messages, please visit
&gt; &lt;https://sourceforge.net/auth/subscriptions/&gt;

**Reply by larspalo** [reply]
Originally posted 2020-07-29 07:34:31.975000 by barbajason in GrandOrgue discussion forum on SourceForge<br><br>On Ubuntu 20.04 succeded with `gunzip -S .cmb &lt;filename&gt;`.
GO v 0.3.1.2334+155 can read the unzipped file.

**Comment by larspalo** [comment]
Originally posted 2020-06-09 09:05:47.115000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Hi Graham

I can also use notepad, or any other text editor to edit OLD .cmb files but recently if you make a NEW .cmb file it is not editable with a text editor because some bright spark has decided to compress the files when they are saved.

I am using GO v0.3.1.2330   It compresses .cmb files when saved but can read both old and new formats.

csw900

**Comment by larspalo** [comment]
Originally posted 2020-06-09 11:31:51.321000 by tbpr in GrandOrgue discussion forum on SourceForge<br><br>I believe in the past that the structure of the .cmb files are not documented because they can change from version to version, and that therefore whatever app or technique you use to deal with can break.

**Comment by larspalo** [comment]
Originally posted 2020-06-09 15:31:35.092000 by edwardstauff in GrandOrgue discussion forum on SourceForge<br><br>Open-source software ought to document all of its file formats.
The reason why I need this is so I can edit the MIDI configuration in a text editor.  Entering it or editing it via the &quot;Midi-settings for ...&quot; dialog is extremely tedious and time consuming.  I&#039;m developing an organ extension, and G.O. has not been very good about remembering the MIDI configuration, and I can&#039;t afford to spend 15–20 minutes re-entering it every time G.O. loses it.

**Reply by larspalo** [reply]
Originally posted 2020-06-09 17:41:31.267000 by [Lars Palo](https://github.com/larspalo) in GrandOrgue discussion forum on SourceForge<br><br>From looking at the code it should be a gzip compression that&#039;s used
(GOrgueCompress.cpp). It&#039;s open source and everyone can read the code.

And yes, your not supposed to manually edit the cmb files that GO creates.
<br><br>Attachments in original post:<br>https://sourceforge.net/p/ourorgan/discussion/962125/thread/e79cea8f67/98af/179d/attachment/alternate<br>

**Reply by larspalo** [reply]
Originally posted 2020-06-21 22:40:13.929000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>OpenSouce means freedom: The freedom, that you can use the program in any way you like; the freedom to redistribute and modify it (in the GO case, the GPL controls modification/distribution with the requirements, that you also need to distribute the GPLed source and all modifications must also be put under the GPL).

OpenSource is not any statement about quality or promise to provide anything specific. 

In regards to keeping settings: Did you save the changes in GO? Can you provide an example, where GO didn&#039;t restore the settings, as they were during the last save?

**Comment by larspalo** [comment]
Originally posted 2020-06-21 22:43:27.817000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>Look at the stable organ file format: You can still load a HW1 file, as its a stable, supported interface.

In contrast, the CMB files are a kind of internal state dump of GO. This might change in any way with any GO change.

**Comment by newVideophile** [comment]
Today, I wanted to edit the MIDI codes that triggers all stops of an organ, and I discovered like the OP that the cmb file does not allow that.  Now I'm going to do this stop by stop, and that's several clicks per stop...

Editable cmd file would be a very a usefule feature indeed !
